### PR TITLE
#16/html structure/layout

### LINKED ISSUE #8: UI | Ustawienie obrazu tła
Kryteria akceptacji

Należy zweryfikować i skonsultować, a także ewentualnie uaktualnić przed rozpoczęciem prac nad zgłoszeniem.

- [x]  Pobrać obraz tła z projektu z Figma.

- [x]  Ustawić obraz jako tło całej aplikacji.

- [x] Tło musi zawsze pokrywać cały ekran, niezależnie od tego ile zajmuje zawartość aplikacji.

### LINKED ISSUE #16: Szkielet aplikacji
- [x] Stworzenie struktury html
- [x] Stworzenie podstawowego layoutu strony

### PR DESCRIPTION
**Pull request by KatarzynaChmielecka**
closes #16 closes #8

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
-          rel="stylesheet">
+<head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>CodersCamp2021 | Harry potter API QUIZ</title>
 </head>
+
 <body>
+    <header>
```
Comment (vieraboschkova) [inline index.html:12]: Perhaps you could wrap header/aside/main/footer in an outside container that would hold the main grid layout

File: index.html [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
 
-<div id="swquiz-app">
-</div>
+    <aside>
+        <div class="game-image"></div>
+    </aside>
 
-<script src="src/index.js"></script>
+    <main>
+        <div class="game-main-content">
```
Comment (vieraboschkova) [inline index.html:24]: Perhaps just game instead of game-main-content ?

Comment (vieraboschkova) [inline index.html:24]: I think you can use <main> to be your "main wrapper". You could move the class to it

File: index.html [hunk 3]
```diff
@@ -1,20 +1,38 @@
...
+            <div class="game-main-content__mode"></div>
+            <div class="game-main-content__mode-rules"></div>
+            <button class="game-main-content__ranking-btn game-main-content__rules-btn"></button>
```
Comment (vieraboschkova) [inline index.html:27]: The two buttons might be easier to style if they come in a separate container. To do that, put them in a div with a class "game__buttons" or similar. Try to use the simplest class names.

File: index.html (hunk 2 above)
Comment (KatarzynaChmielecka) [reply index.html:24]: Probably it's a 'little' mess caused by an earlier version :)

File: index.html (hunk 3 above)
Comment (KatarzynaChmielecka) [reply index.html:27]: I don't know why I didn't think about it earlier :) It's so obvious... ;)

File: index.html [hunk 4]
```diff
@@ -1,20 +1,39 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>CodersCamp2021 | Harry potter API QUIZ</title>
+    <link rel="stylesheet" href="./styles/index.css">
 </head>
+
 <body>
+    <div class="game-wrapper">
+        <header class="header-content">
```
Comment (vieraboschkova) [inline index.html:14]: I think we can remove content from the classname also here, what do you think?

File: index.html [hunk 5]
```diff
@@ -1,20 +1,39 @@
...
+        <aside class="game-image">
 
-<script src="src/index.js"></script>
+        </aside>
+
+        <main class="game">
+            <div class="game__mode"></div>
+            <div class="game__mode-rules"></div>
+            <div class="game__btns">
+                <button class="game__ranking game__rules"></button>
```
Comment (vieraboschkova) [inline index.html:27]: They - both buttons will probably have the same style, although different content. But that doesnt mean they will nead seaprete styling. I think for now you can create the same class for both buttons - game__btn or game__button

File: index.html [hunk 6]
```diff
@@ -1,20 +1,39 @@
...
+                <button class="game__play"></button>
```
Comment (vieraboschkova) [inline index.html:28]: We could consider adding #ids to the buttons, we will probably need to get them from js side. We can use querySelector also thuogh, but we can cosider ids.

File: index.html [hunk 7]
```diff
@@ -1,20 +1,39 @@
...
+            </div>
+        </main>
+
+        <footer class="footer">
+            <button class="game-settings"></button>
```
Comment (vieraboschkova) [inline index.html:33]: I would give him the same styling as other buttons for now - game__button

File: index.html [hunk 8]
```diff
@@ -1,20 +1,39 @@
...
+            <div class="game-time-lapse"></div>
```
Comment (vieraboschkova) [inline index.html:34]: game__timer perhaps?

File: styles/index.scss [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  font-family: 'Bitter', serif;
+}
+
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+//50 85 25
+.game-wrapper {
```
Comment (vieraboschkova) [inline styles/index.scss:11]: I think we don't need to connect all the classes with the .game-wrapper, it compiles to: ".game-wrapper .header-content". I think they  can work as single classes without relation to the wrapper, then you could use the scss in a simpler way (less lines of code):
`
  .game {
    // costam
    &-wrapper {
      // costam
    }
    &-image {
      // costam
    }
  }
  .footer {
    // costam
  }
  .header {
    // costam
  }`

File: styles/index.scss [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  background-image: url('../images/QuizBackground.png');
```
Comment (vieraboschkova) [inline styles/index.scss:12]: Background loads full screen and the grid divides well, good job!

File: styles/index.scss [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  grid-template-columns: 1fr 2fr;
+  //   grid-template-rows: 30% 55% 15%;
+  grid-template-rows: 2fr 4fr 1fr;
+  .header-content {
+    background: burlywood;
+    grid-column: 1/3;
+    grid-row: 1/2;
+  }
+  .game-image {
+    background: cadetblue;
```
Comment (vieraboschkova) [inline styles/index.scss:27]: I usually test using something like: `outline: cadetblue 10px dotted`, just an idea :)

File: index.html (hunk 4 above)
Comment (KatarzynaChmielecka) [reply index.html:20]: Of course :) I forgot about it.

File: index.html (hunk 5 above)
Comment (KatarzynaChmielecka) [reply index.html:33]: You're right :)

File: index.html (hunk 6 above)
Comment (KatarzynaChmielecka) [reply index.html:34]: Ok, it's not a problem :)

File: index.html (hunk 7 above)
Comment (KatarzynaChmielecka) [reply index.html:39]: I'm not sure about it, but ok :)

File: index.html (hunk 8 above)
Comment (KatarzynaChmielecka) [reply index.html:40]: Yeees, I missed that word :)

File: styles/index.scss (hunk 9 above)
Comment (KatarzynaChmielecka) [reply styles/index.scss:11]: Still, I don't know what are you talking about, we should have a call probably :)

File: styles/index.scss (hunk 11 above)
Comment (KatarzynaChmielecka) [reply styles/index.scss:27]: Yes, it's a better way :) Just I forgot about the outline... ;)

File: index.html [hunk 12]
```diff
@@ -1,21 +1,44 @@
...
+        <main class="game">
+            <div class="game__mode"></div>
+            <div class="game__mode-rules"></div>
+            <div class="game__btns">
+                <button class="game__btn" id="game__btn"></button>
+                <button class="game__play" id="game_play"></button>
+            </div>
+        </main>
 
-<script type="module" src="src/index.js"></script>
```
Comment (vieraboschkova) [inline index.html:19]: The script will be needed, I think it's best to keep it.

File: index.html [hunk 13]
```diff
@@ -1,21 +1,44 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link
+        href="https://fonts.googleapis.com/css2?family=Bitter:ital,wght@0,400;0,500;0,800;1,400;1,500;1,600;1,800&display=swap"
+        rel="stylesheet">
 </head>
+
 <body>
+    <div class="game-wrapper">
+        <header class="header">
+            <div class="header__game-logo"></div>
```
Comment (vieraboschkova) [inline index.html:20]: perhaps just header__logo?

File: index.html [hunk 14]
```diff
@@ -1,21 +1,44 @@
...
+    <link rel="stylesheet" href='./static/styles/index.css'>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link
+        href="https://fonts.googleapis.com/css2?family=Bitter:ital,wght@0,400;0,500;0,800;1,400;1,500;1,600;1,800&display=swap"
+        rel="stylesheet">
 </head>
+
 <body>
+    <div class="game-wrapper">
```
Comment (vieraboschkova) [inline index.html:18]: game__wrapper perhaps?

File: index.html [hunk 15]
```diff
@@ -1,21 +1,44 @@
...
+            <div class="game__btns">
+                <button class="game__btn" id="game__btn"></button>
+                <button class="game__play" id="game_play"></button>
+            </div>
+        </main>
 
-<script type="module" src="src/index.js"></script>
+        <footer class="footer">
+            <button class="game__btn" id="game__btn"></button>
+            <div class="game-timer"></div>
```
Comment (vieraboschkova) [inline index.html:39]: game__timer maybe?

File: index.html [hunk 16]
```diff
@@ -1,21 +1,44 @@
...
+            <div class="game-image__content"></div>
+        </aside>
 
-</div>
+        <main class="game">
+            <div class="game__mode"></div>
+            <div class="game__mode-rules"></div>
+            <div class="game__btns">
+                <button class="game__btn" id="game__btn"></button>
+                <button class="game__play" id="game_play"></button>
```
Comment (vieraboschkova) [inline index.html:33]: This one will probably use the general game__btn styling also.

File: static/styles/index.scss [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+    align-items: center;
+
+    &__content {
+      width: 60%;
+      height: 60%;
+      outline: yellow dotted 5px;
+    }
+  }
+
+  .game {
```
Comment (vieraboschkova) [inline static/styles/index.scss:53]: You could put &-image and &-wrapper also inside in the general .game bracket, the .game-wrapper class doesnt need to wrap all the other classes.

**Review by vieraboschkova — APPROVED** [review]
Good job! I think the structure is more clear now. Thank you!

**Pull request merged**